### DISCUSSION TITLE
KeystoreExplorer GUI

### DISCUSSION THREAD

**Discussion by stokito** [ebourg/jsign#182] [General]
The [KeystoreExplorer](https://keystore-explorer.org/) is a Java app that allows to create a JKS or P12 keystore, generate key pair, manage certs and it has a built-in functions to sign a jar file.

Maybe you can collaborate with the project and add more signing tools. This can be a basic UI for the jsign. Maybe some common parts can be extracted to a separate library or even added to the BouncyCastle to be reused by other projects.
I was thinking about to add to the KSE an ability to sign regular files with a detached `.p7s` sign but decided to search for existing tools first and found your project.

It's currently focused on code signing while I more interested in a more broad usage for regular users.
For example my government allows to use the PKCS#7 signatures to sign contracts but there is no an official GUI for Linux and macOS. So I think to use KSE for this.

**Comment by ebourg** [comment]
I'm definitely interested in collaborating with other projects, let me know if the is something specific I can do to help integrating Jsign with KSE.

The Jsign API exposes a JCE provider that could be used by KSE to access cloud based keystores or hardware keystores (Yubikey, Nitrokey, PGP cards, PKCS#11 tokens, etc). At some point I'll probably spin this off as an independent component or project.
